### PR TITLE
Allow to saturate counts

### PR DESCRIPTION
**Pull request by efernandez**
refs CORE-5273

@servos @afakihcpr

### CODE REVIEW

File: src/grid_overlay.cpp [hunk 1]
```diff
@@ -216,6 +225,27 @@ void resetCounts (OverlayClouds* overlay)
...
+  for (size_t i = 0; i < overlay.hit_counts.size(); ++i)
+  {
+    const unsigned hit_count = overlay.hit_counts[i];
+    const unsigned pass_through_count = overlay.pass_through_counts[i];
+
+    const double occupancy_threshold = overlay.occupancy_threshold;
+    const double hit_threshold = pass_through_count * occupancy_threshold;
+
+    overlay.hit_counts[i] = hit_count > hit_threshold ?
+                            hit_threshold + steps + 1 :
```
Comment (efernandez) [inline src/grid_overlay.cpp:244]: Note that `hit_threshold + steps` is actually a simplification of `(pass_through_count + steps / occupancy_threshold) * occupancy_threshold`, where `steps / occupancy_threshold` is indeed `steps * increment(occupancy_threshold)`

File: src/grid_overlay.cpp [hunk 2]
```diff
@@ -216,6 +225,27 @@ void resetCounts (OverlayClouds* overlay)
...
+                            std::max(hit_threshold - steps, 0.0);
```
Comment (efernandez) [inline src/grid_overlay.cpp:245]: This is needed because `determineOccupancy` takes `unsigned` args! :see_no_evil:

**Comment by servos** [comment]
LGTM.

File: src/grid_overlay.cpp [hunk 3]
```diff
@@ -113,7 +122,7 @@ void addKnownFreePoint (OverlayClouds* overlay, const gm::Point& p, const double
       {
         const index_t ind = cellIndex(geom, c2);
         overlay->hit_counts[ind] = 0;
-        overlay->pass_through_counts[ind] = overlay->min_pass_through+1;
+        overlay->pass_through_counts[ind] = overlay->min_pass_through * increment(overlay.occupancy_threshold) + 1;
```
Comment (efernandez) [inline src/grid_overlay.cpp:125]: Just check all places where `pass_through_counts` is used and updated/scaled the `min_pass_through` value.

**Pull request merged**